dbInterface.js

```diff
@@ -83,6 +85,29 @@ class DbInterface {
       return false;
     }
   }
+
+  async fetchSalt(username) {
+    let response = {
+      success: false,
+      salt: '',
+    };
+    const text = `
+      SELECT salt
+      FROM user_auth.auth
+      WHERE username=$1;
+    `;
+    const values = [username];
+    try {
```

Comment:
You're going to be so happy once I teach you how to do a catch-all for all of these, but your architecture doesn't support it yet :D